storage/src/main/java/tech/pegasys/teku/storage/server/ReplayDepositsResult.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.storage.server;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.Optional;
+
+public class ReplayDepositsResult {
+  private final Optional<UnsignedLong> blockNumber;
+  private final boolean pastGenesisBlock;
+
+  public ReplayDepositsResult(final UnsignedLong blockNumber, final boolean pastGenesisBlock) {
+    this.blockNumber = Optional.ofNullable(blockNumber);
+    this.pastGenesisBlock = pastGenesisBlock;
+  }
+
+  public Optional<UnsignedLong> getBlockNumber() {
+    return blockNumber;
+  }
+
+  public boolean isPastGenesisBlock() {
+    return pastGenesisBlock;
```

Comment:
nit: rename to pastMinGenesisBlock - it's the block that meets the timestamp criteria but may not actually trigger genesis as it may not have enough validators.